lib/pairing.js

```diff
@@ -16,15 +17,23 @@ function pullToken (token) {
   return db.one(sql, [token])
 }
 
+// TODO new-admin: We should remove all configs related to that device. This can get tricky.
 function unpair (deviceId) {
-  // TODO new-admin: We should remove all configs related to that device. This can get tricky.
-  return db.tx(t => {
-    const q1 = t.none('DELETE FROM devices WHERE device_id=$1', [deviceId])
-    const q2 = t.none('DELETE FROM machine_pings WHERE device_id=$1', [deviceId])
-    const q3 = t.none('DELETE FROM machine_network_heartbeat WHERE device_id=$1', [deviceId])
-    const q4 = t.none('DELETE FROM machine_network_performance WHERE device_id=$1', [deviceId])
-    return Promise.all([q1, q2, q3, q4])
-  })
+  const sql = `INSERT INTO
+    unpaired_devices(id, device_id, name, model, paired, unpaired)
+    VALUES (
+      $1,
+      $2,
+      (SELECT name FROM devices WHERE device_id=$2),
+      (SELECT model FROM devices WHERE device_id=$2),
+      (SELECT created FROM devices WHERE device_id=$2),
+      now());
+    DELETE FROM devices WHERE device_id=$2;
+    DELETE FROM machine_pings WHERE device_id=$2;
+    DELETE FROM machine_network_heartbeat WHERE device_id=$2;
+    DELETE FROM machine_network_performance WHERE device_id=$2;`
+
+  return  db.tx(t => t.none(sql, [uuid.v4(), deviceId]))
```

Comment:
It's better to keep the individual queries separated, both to simplify versioning control and avoid query clumping, as we already have on some other functions. The pre-commit format works great, apart from not calling `t.batch` on the return value

```suggestion
  const q1 = t.none(`INSERT INTO
    unpaired_devices(id, device_id, name, model, paired, unpaired)
    VALUES (
      $1,
      $2,
      (SELECT name FROM devices WHERE device_id=$2),
      (SELECT model FROM devices WHERE device_id=$2),
      (SELECT created FROM devices WHERE device_id=$2),
      now())`, [uuid.v4(), deviceId])
  const q2 = t.none(`DELETE FROM devices WHERE device_id=$1`, [deviceId])
  const q3 = t.none(`DELETE FROM machine_pings WHERE device_id=$1`, [deviceId])
  const q4 = t.none(`DELETE FROM machine_network_heartbeat WHERE device_id=$1`, [deviceId])
  const q5 = t.none(`DELETE FROM machine_network_performance WHERE device_id=$1`, [deviceId])

  return t.batch([q1, q2, q3, q4, q5])
```

Reply:
Good points. I can't directly accept your suggestion as `t.batch` is doesn't guarantee that DELETE queries will be run after `q1` is completed (that is main reason why I put everything in one query).